### PR TITLE
Initial work

### PR DESCRIPTION
**Pull request by humphd**

### CODE REVIEW

**Comment by humphd** [comment]
@alicoding, can you review this for me?  It will allow us to use a db server to store the data.  I want to use it to test different options (postgres vs. mysql vs. s3) in production.

**Comment by alicoding** [comment]
Sure I will review and test this

File: README.md [hunk 1]
```diff
@@ -1,4 +1,47 @@
...
+  type: SQLProvider.MYSQL,
+
+  // If you want SQL debug logging, pass a logging function
+  //logging: console.log,
+
+  // A unique string to identify this user's filesystem (e.g., username)
+  user: 'something-unique',
+
+  // Database options
+  db: {
```
Comment (alicoding) [inline README.md:36]: Lets also add option to pass in `url` since sequilize supports that too

**Comment by alicoding** [comment]
I'm trying to do `writeFile()` using demo page in MakeDrive and server throws these errors

```
[2014-10-14T15:16:44.645Z] ERROR: MakeDrive/20378 on ali.local: rsync.patch() error
    err: {
      "name": "SequelizeUniqueConstraintError",
      "parent": {
        "errno": 19,
        "code": "SQLITE_CONSTRAINT",
        "sql": "INSERT INTO `makedrive_data` (`user`,`key`,`data`) VALUES ('ali','1CAA2BE8-6A9D-4118-A054-D4CDAB6DED36',X'7b226e616d65616263223a7b226964223a2232383337313733462d443331412d343642322d424643452d333930313243343938324235222c2274797065223a2246494c45227d7d');"
      },
      "sql": "INSERT INTO `makedrive_data` (`user`,`key`,`data`) VALUES ('ali','1CAA2BE8-6A9D-4118-A054-D4CDAB6DED36',X'7b226e616d65616263223a7b226964223a2232383337313733462d443331412d343642322d424643452d333930313243343938324235222c2274797065223a2246494c45227d7d');",
      "fields": [
        "user",
        "key"
      ],
      "value": null,
      "index": null
    }
    --
    client: {
      "username": "ali",
      "id": "aa260f71-c43b-462e-9766-54055280400b",
      "state": "LISTENING",
      "path": "/nameabc"
    }
[2014-10-14T15:16:44.648Z] ERROR: MakeDrive/20378 on ali.local: Client should own lock but does not. (err={})
```

**Comment by alicoding** [comment]
I'm testing with sqlite right now

**Comment by alicoding** [comment]
Just tested with mysql

```
MakeDrive Filesystem Initialization Error:  [Error: Error: ER_BAD_DB_ERROR: Unknown database 'makedrive']
```

This is my first time connecting using mysql, so I had this error, but once I created my own database with the name `makedrive` then I can proceed.

**Comment by humphd** [comment]
I'll document in the README that you should create your DB manually.

**Comment by humphd** [comment]
I've added the connection URL option.  Can you test that, too?

**Comment by alicoding** [comment]
r=me. Tested all the DBs and it's working great.

**Pull request merged**